Source/WebCore/dom/Element.h

```diff
@@ -783,6 +783,9 @@ class Element : public ContainerNode {
     bool capturedInViewTransition() const { return hasElementStateFlag(ElementStateFlag::CapturedInViewTransition); }
     void setCapturedInViewTransition(bool captured) { setElementStateFlag(ElementStateFlag::CapturedInViewTransition, captured); }
 
+    void setWritingSuggestion(const String&);
+    String writingSuggestion() const;
+
```

Comment:
Are there going to be multiple elements with writing suggestions at the same time? If not why does this need to be an Element thing rather than a Document level concept (like Editor and FrameSelection)?

Reply:
how would I access Editor or FrameSelection from within `RenderTreeUpdater::GeneratedContent`?

Reply:
They are both available in Document which can be found via Node::document() for any Node/Element.

Reply:
I moved the writing suggestion data to Editor